boot/seal.go

```diff
@@ -150,7 +179,17 @@ func resealKeyToModeenv(model *asserts.Model, modeenv *Modeenv) error {
 
 	pbc := toPredictableBootChains(append(runModeBootChains, recoveryBootChains...))
 
-	// TODO:UC20: load and compare the predictable bootchains
+	ok, nextCount, err := isResealNeeded(pbc, rootdir)
+	if err != nil {
+		return err
+	}
+	if !ok {
+		// no need to actually reseal
+		logger.Debugf("reseal not necessary")
+		return nil
+	}
+	pbcJSON, _ := json.Marshal(pbc)
```

Comment:
why not check the error here?

Reply:
Debugf can take the nil and pbcJSON is built only for Debugf here. We also do the same encoding operation before here when we decide whether to reseal and later when we write the pbc out, so a problem wouldn't be missed.

Reply:
If Debugf had a %j modifier that did the encoding for us, it would also not result in a propagated error.